### PR TITLE
fixing belongs_to :meta_thing, :dependent => destroy, polymorphic => true

### PR DESCRIPTION
**Pull request by taelor**
Instead of trying to constantize the name of the relationship (i.e. meta_thing) by calling reflection.klass (and possibly getting a bug for unknown class Parent::MetaThing), we can look to the reflection's foreign_type on polymorphic associations, and constantize that to use for determining the scope when recovering and destroying deleted_at objects.

All the tests passed, and it helped fix our use case for this gem.

### CODE REVIEW

File: lib/acts_as_paranoid/core.rb [hunk 1]
```diff
@@ -176,6 +176,14 @@ def deleted?
     alias_method :destroyed?, :deleted?
 
     private
+    
+    def get_reflection_class(reflection)
+      if reflection.macro == :belongs_to && reflection.options.include?(:polymorphic)
```
Comment (chuckg) [inline lib/acts_as_paranoid/core.rb:181]: What situation does this cover? Would it be possible to get a test case to more clearly cover it? 

Everything else looks good.

Comment (taelor) [inline lib/acts_as_paranoid/core.rb:181]: It's kind of a crazy case. We have a pretty complex structure due to some legacy promises. I'll see if I can write some tests for it, but let me semi explain it here.

```
class Parent < ActiveRecord::Base
  acts_as_paranoid
  has_many :sections, :dependent => destroy

class Section < ActiveRecord::Base
  acts_as_paranoid
  belongs_to :meta_thing, :polymorphic => true, :dependent => destroy

class SomeThing < ActiveRecord::Base
  acts_as_paranoid
  has_one :section, :as => :meta_thing

class AnotherThing < ActiveRecord::Base
  acts_as_paranoid
  has_one :section, :as => :meta_thing
```

So in this situation, we want to delete, then recover the Parent objects. But when you delete/recover, we also want it to propagate down. The problem happens when you you try and call `reflection.klass` when looking at the reflection for Section belongs_to :meta thing, reflection.klass tries to constantize Section::MetaThing, but that class doesn't actually exist.

Instead, we can look at the foreign_type, which should be the column "meta_thing_type", and would hold a value of either "SomeThing" or "AnotherThing". So in reflection_class() we are calling `section.meta_thing_type.constantize`, giving us one of the two child classes, SomeThing or AnotherThing

Comment (taelor) [inline lib/acts_as_paranoid/core.rb:181]: OK, so I've added some tests for you. I wasn't sure how to get that second commit added to the pull request, but boom, when I refreshed it was there. Sweet.

Comment (taelor) [inline lib/acts_as_paranoid/core.rb:181]: Let me know if there is anything else I need to do to help out, or to get this approved.

**Comment by leonid-shevtsov** [comment]
Could this get merged to the master any time soon? I can confirm that the fix works.

**Comment by mhuggins** [comment]
Is this repo not active anymore?

**Comment by taelor** [comment]
@goncalossilva can you give this a look? I wrote a test for the given situation, and it seems to be helping some others as well. If you need anything else for me to do to get this accepted, just let me know. Thanks!

**Comment by goncalossilva** [comment]
@taelor Looks good, thanks!

Will release a new version in the next few days with this fix.

**Comment by mhuggins** [comment]
I guess that answers my question. :)  Glad to see this is still active!

**Comment by goncalossilva** [comment]
@mhuggins truth is that I'm not currently using this on any app in production, so that makes it a bit harder to find time to work on it. Abandoned? No, but it definitely needs some love :)

**Pull request merged**